### DISCUSSION TITLE
AppThemeBinding in Markup

### DISCUSSION THREAD

**Discussion by fredericoregateiro** [CommunityToolkit/Maui#627] [Q&A] [answered]
Hello,

Is there a way to use the AppThemeBinding when creating a style in code?

Thanks

**Comment by bijington** [comment, marked as answer]
I believe we have gone with the approach in this PR when supporting light/dark mode:

https://github.com/CommunityToolkit/Maui.Markup/pull/98

**Reply by TheCodeTraveler** [reply]
Yup! But we are certainly open to better ideas + implementations if you have them, @fredericoregateiro!

**Reply by fredericoregateiro** [reply]
I was waiting for something more "official", for now i'm using this extension method:

```csharp
public static V AppThemeBinding<V, T>(this V view, BindableProperty bindable, T light, T dark) where V : BindableObject
        {
            view.SetAppTheme(bindable, light, dark);

            return view;
        }
```

and call on the object
```csharp
new Button().AppThemeBinding(Button.TextColorProperty, Colors.Green, Colors.Orange)
```

I haven't had more time to do something better

**Reply by bijington** [reply]
> I was waiting for something more "official", for now i'm using this extension method:
> 
> 
> 
> ```csharp
> 
> public static V AppThemeBinding<V, T>(this V view, BindableProperty bindable, T light, T dark) where V : BindableObject
> 
>         {
> 
>             view.SetAppTheme(bindable, light, dark);
> 
> 
> 
>             return view;
> 
>         }
> 
> ```
> 
> 
> 
> and call on the object
> 
> ```csharp
> 
> new Button().AppThemeBinding(Button.TextColorProperty, Colors.Green, Colors.Orange)
> 
> ```
> 
> 
> 
> I haven't had more time to do something better

I like this approach. I think it ties in with something we had discussed previously but I am struggling to find it now. @brminnick what do you think to this? Perhaps we could open a proposal for it?

**Reply by TheCodeTraveler** [reply]
Genius! I love this idea and it'd be great @fredericoregateiro if you could [submit a Proposal](https://github.com/CommunityToolkit/Maui.Markup/issues/new?assignees=&labels=new%2C+proposal&template=open-a-new-feature-proposal.md&title=%5BProposal%5D+) for it 💯 

Maybe we could add a second method for [`.SetAppThemeColor`](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/system-theme-changes#extension-methods) in the same Proposal too!

**Reply by fredericoregateiro** [reply]
Thanks @brminnick and @bijington  for the feedback, I'll take care of the proposal

**Comment by CliffAgius** [comment]
Closed as answered
